p2p/host.go

```diff
@@ -70,24 +64,29 @@ func NewHost(self *Peer, key libp2p_crypto.PrivKey) (Host, error) {
 		return nil, errors.Wrapf(err,
 			"cannot create listen multiaddr from port %#v", self.Port)
 	}
+
 	ctx := context.Background()
 	p2pHost, err := libp2p.New(ctx,
-		libp2p.ListenAddrs(listenAddr), libp2p.Identity(key),
+		libp2p.ListenAddrs(listenAddr),
+		libp2p.Identity(key),
+		libp2p.EnableNATService(),
+		libp2p.ForceReachabilityPublic(),
 	)
 	if err != nil {
 		return nil, errors.Wrapf(err, "cannot initialize libp2p host")
 	}
-	traceFile := os.Getenv("P2P_TRACEFILE")
 
-	const MaxSize = 2_145_728
 	options := []libp2p_pubsub.Option{
-		libp2p_pubsub.WithPeerOutboundQueueSize(64),
-		libp2p_pubsub.WithMaxMessageSize(MaxSize),
-	}
-	if len(traceFile) > 0 {
-		tracer, _ := libp2p_pubsub.NewJSONTracer(traceFile)
-		options = append(options, libp2p_pubsub.WithEventTracer(tracer))
+		// WithValidateQueueSize sets the buffer of validate queue. Defaults to 32. When queue is full, validation is throttled and new messages are dropped.
+		libp2p_pubsub.WithValidateQueueSize(64),
```

Comment:
Shall we make this bigger? The validation has sig verification and costly operations. I suspect many msg may be dropped if it's leader. Also, if there a log when msg gets dropped due to validation throttle?

Reply:
We don't do sign verification on the validation anymore.  Based on my benchmarking, sig verification will triple the CPU usage on a regular node, thus it will cause a lot of latency during attack.  So I reverted it back to original logic to do late sig verification.

As to the log, I think the p2p trace will have it, but we didn't enough the p2p trace by default though.

Reply:
The signature verification is done anyway if attacker send a lot of msg when a fake sender public key that's in the committee. But for attacker who don't use fake key in the committee, the delayed sig verification will help.